plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -128,6 +128,17 @@ func isVolumeConflict(volume api.Volume, pod *api.Pod) bool {
 			}
 		}
 
+		if volume.ISCSI != nil && existingVolume.ISCSI != nil {
+			iqn, lun, target := volume.ISCSI.IQN, volume.ISCSI.Lun, volume.ISCSI.TargetPortal
+			eiqn, elun, etarget := existingVolume.ISCSI.IQN, existingVolume.ISCSI.Lun, existingVolume.ISCSI.TargetPortal
+			// two ISCSI volumes are same, if they share the same iqn, lun and target. As iscsi volumes are of type
+			// RWO or ROX, we could permit only one RW mount. Same iscsi volume mounted by multiple Pods
+			// conflict unless all other pods mount as read only.
+			if iqn == eiqn && lun == elun && target == etarget && !(volume.ISCSI.ReadOnly && existingVolume.ISCSI.ReadOnly) {
```

Comment:
there is some tricks with lun and target defaults . LUN default is 0 and Target could be in form of `ip:port` or `ip` with default port 3260, if not provided. Both cases have to be evaluated.